environment/src/main/java/com/sequenceiq/environment/credential/v1/AuditCredentialV1Controller.java

```diff
@@ -0,0 +1,106 @@
+import com.sequenceiq.environment.credential.v1.converter.CredentialToCredentialV1ResponseConverter;
+import com.sequenceiq.notification.NotificationController;
+
+@Controller
+@InternalReady
+@AuthorizationResource
+public class AuditCredentialV1Controller extends NotificationController implements AuditCredentialEndpoint {
+
+    private final CredentialService credentialService;
+
+    private final CredentialToCredentialV1ResponseConverter credentialConverter;
+
+    public AuditCredentialV1Controller(
+            CredentialService credentialService,
+            CredentialToCredentialV1ResponseConverter credentialConverter) {
+        this.credentialService = credentialService;
+        this.credentialConverter = credentialConverter;
+    }
+
+    @Override
+    @FilterListBasedOnPermissions(action = AuthorizationResourceAction.DESCRIBE_CREDENTIAL)
+    public CredentialResponses list() {
+        String accountId = ThreadBasedUserCrnProvider.getAccountId();
+        return new CredentialResponses(
+                credentialService.listAvailablesByAccountId(accountId, onlyAudit())
+                        .stream()
+                        .map(credentialConverter::convert)
+                        .collect(Collectors.toSet()));
+    }
+
+    @Override
+    @CheckPermissionByResourceCrn(action = AuthorizationResourceAction.DESCRIBE_CREDENTIAL)
+    public CredentialResponse getByResourceCrn(@TenantAwareParam @ResourceCrn String credentialCrn) {
+        String accountId = ThreadBasedUserCrnProvider.getAccountId();
+        return credentialConverter.convert(credentialService.getByCrnForAccountId(credentialCrn, accountId, onlyAudit()));
+    }
+
+    @Override
+    @CheckPermissionByAccount(action = AuthorizationResourceAction.CREATE_CREDENTIAL)
+    public CredentialResponse post(@Valid CredentialRequest request) {
+        String accountId = ThreadBasedUserCrnProvider.getAccountId();
+        String creator = ThreadBasedUserCrnProvider.getUserCrn();
+        Credential credential = credentialConverter.convert(request);
+        credential.setType(AUDIT);
+        notify(ResourceEvent.CREDENTIAL_CREATED);
+        Set<Credential> auditCredentialsByPlatfom = credentialService
+                .listAvailablesByAccountId(accountId, onlyAudit())
+                .stream()
+                .filter(c -> c.getCloudPlatform().equals(credential.getCloudPlatform()))
+                .collect(Collectors.toSet());
+        if (auditCredentialsByPlatfom.isEmpty()) {
+            return credentialConverter.convert(credentialService.create(credential, accountId, creator, onlyAudit()));
+        } else {
+            throw new BadRequestException(String.format("Audit credential already exist for %s cloud.", credential.getCloudPlatform()));
+        }
+    }
+
+    @Override
+    @CheckPermissionByAccount(action = AuthorizationResourceAction.CREATE_CREDENTIAL)
+    public CredentialResponse put(@ResourceObject @Valid CredentialRequest credentialRequest) {
```

Comment:
this should be authorized against the existing credential resource, not on account level
to do that:
- add `@CheckPermissionByResourceObject` to the method
- add `@ResourceObject` to the credentialrequest parameter (it is there as i see)
- add `@ResourceObjectField(action = AuthorizationResourceAction.EDIT_CREDENTIAL, variableType = AuthorizationVariableType.NAME)` to the `name` field of the CredentialRequest class (I think this is done too since it is the same class as used in other crednetial controller)

so I think the method annotation should be changed only